docs/contributor/writing-tests-for-vscode-swift.md

```diff
@@ -113,6 +114,35 @@ suite("ReloadExtension Unit Test Suite", () => {
 
 You may have also noticed that we needed to cast the `"Reload Extensions"` string to `any` when resolving `showWarningMessage()`. Unforunately, this may be necessary for methods that have incompatible overloaded signatures due to a TypeScript issue that remains unfixed.
 
+## Mocking the File System
+
+Mocking file system access can be a challenging endeavor that is prone to fail when implementation details of the unit under test change. This is because there are many different ways of accessing and manipulating files, making it almost impossible to catch all possible failure paths. For example, you could check for file existence using `fs.stat()` or simply call `fs.readFile()` and catch errors with a single function call. Using the real file system is slow and requires extra setup code in test cases to configure.
```

Comment:
For pattern where you want to test when the fs functions throws do we use the mockGlobalModule route?

Reply:
To test a file not existing you can just create an empty file system which will cause `fs.stat()` or `fs.readFile()` to throw an error. You could also set the mode on the created file using `mockFS.file({ mode: 0o000 })` so that it's not readable/writable by the current user.

Reply:
Good question, though. I'm going to add an example for this.

Reply:
Yeah, because to mock it to throw is like 2 lines. But I think what you said is better as it's more explicit about the expectation around fs and we should establish as such.